core/renderer/backend/opengl/RenderTargetGL.cpp

```diff
@@ -44,40 +44,45 @@ void RenderTargetGL::unbindFrameBuffer() const
 }
 
 void RenderTargetGL::update() const {
-    if (!_dirty) return;
-    if(!_defaultRenderTarget) {
-        { // color attachments
+    if (!_dirtyFlags) return;
+    if (!_defaultRenderTarget)
+    {
+        {  // color attachments
             GLenum bufs[MAX_COLOR_ATTCHMENT] = {GL_NONE};
             for (size_t i = 0; i < MAX_COLOR_ATTCHMENT; ++i)
             {
-                GLuint texture = 0;
-                GLint level = 0;
-                if (_color[i])
-                {
-                    texture = static_cast<GLuint>(_color[i].texture->getHandler());
-                    level = _color[i].level;
-                    bufs[i] = GL_COLOR_ATTACHMENT0 + i;
-                }
-                glFramebufferTexture2D(GL_FRAMEBUFFER, GL_COLOR_ATTACHMENT0 + i, GL_TEXTURE_2D, texture, level);
+                if (!bitmask::any(_dirtyFlags, getMRTColorFlag(i)))
+                    continue;
+                auto textureInfo = _color[i];
+                auto textureHandler =
+                    static_cast<GLuint>(textureInfo.texture != nullptr ? textureInfo.texture->getHandler() : 0);
+                glFramebufferTexture2D(GL_FRAMEBUFFER, GL_COLOR_ATTACHMENT0 + i, GL_TEXTURE_2D, textureHandler,
+                                       textureInfo.level);
+                bufs[i] = GL_COLOR_ATTACHMENT0 + i;
             }
-    #if AX_TARGET_PLATFORM == AX_PLATFORM_WIN32 || AX_TARGET_PLATFORM == AX_PLATFORM_LINUX
+#if AX_TARGET_PLATFORM == AX_PLATFORM_WIN32 || AX_TARGET_PLATFORM == AX_PLATFORM_LINUX
             glDrawBuffers(MAX_COLOR_ATTCHMENT, bufs);
```

Comment:
It's also possible to skip this call if no color attachments were changed. Not sure it's worth it though.